aya/src/programs/lirc_mode2.rs

```diff
@@ -118,20 +120,23 @@ pub struct LircLinkId(RawFd, RawFd);
 /// An LircMode2 Link
 pub struct LircLink {
     prog_fd: RawFd,
```

Comment:
I am concerned about the inconsistency that'd create since every other link does not have an owned program (it only owns the target). I think we need a separate PR to resolve the file descriptor safety in the program fd's inside links.